src/client-config.ts

```diff
@@ -0,0 +1,18 @@
+import { type, release } from 'os';
+import { readFileSync } from 'fs';
+import { env } from 'vscode';
+
+const CLIENT_NAME = 'Monokle VSCode';
+
+export function getClientConfig() {
+	const pkg = JSON.parse(readFileSync(`${__dirname}/../package.json`, 'utf8'));
+
+	return {
+		name: CLIENT_NAME,
+		version: pkg.version,
+		os: `${type()} ${release()}`,
+		additionalData: {
+			machineId: env.machineId,
```

Comment:
For privacy concerns, maybe we should only add this when analytics are enabled?

Reply:
Makes sense, especially that without telemetry we won't get any other data to connect with this one by machineId.